File: Code Samples/stack.py
class Stack(object) :
	def __init__(self):
		self.top = 0
		self.items = []

	def __getitem__(self,key):
		return self.items[key]

	def push(self, item):
		self.top += 1
		self.items.append(item)

	def pop(self):
		self.top -= 1
		return self.items.pop()

	def isEmpty(self):
		return self.top == 0




# Tests
s = Stack()
s.push(4)
s.push(5)

assert s.items == [4,5]
assert not s.isEmpty()
assert s[1]    == 5
assert s.pop() == 5
assert s.pop() == 4
assert s.isEmpty()

File: Code Samples/Stack.java
import java.util.Arrays;

class Stack<Type> {
	public int size = 0;
	private Object items[];
	private static final int DEFAULT_CAPACITY = 10;

	public Stack(){
		items = new Object[DEFAULT_CAPACITY];
	}
	private void resize(){
		int newSize = items.length * 2;
		items = Arrays.copyOf(items, newSize);
	}
	public void push(Type item){
		if(size == items.length){
			resize();
		}
		items[size++] = item;
	}
	public Type pop() {
		Type item = (Type) items[--size];
		items[size] = null;
		return item;
	}
	public Boolean isEmpty(){
		return size == 0;
	}



	/* TESTS */
	public static void main(String[] args) {
		Stack s = new Stack<Integer>();

		s.push(4);
		s.push(5);

		assert !s.isEmpty();
		assert s.pop() == 5;
		assert s.pop() == 4;
		assert s.isEmpty();
	}
}
